src/components/Editor/SearchBar.js

```diff
@@ -262,11 +264,9 @@ const SearchBar = React.createClass({
       return;
     }
 
-    const symbolDeclarations = getSymbols(
-      sourceText.toJS())[selectedSymbolType];
-
+    const symbolDeclarations = await getSymbols(sourceText.toJS());
```

Comment:
Turns out trying to access a member of something you're awaiting *immediately* can cause undefined returns. Moving the access to afterwards fixed the issue of `getSymbols` returning  undefined.